src/parallel/executor.cpp

```diff
@@ -468,6 +470,10 @@ PendingExecutionResult Executor::ExecuteTask() {
 		if (!task) {
 			scheduler.GetTaskFromProducer(*producer, task);
 		}
+		if (!task && !HasError() && !to_be_rescheduled_tasks.empty()) {
```

Comment:
`to_be_rescheduled_tasks` needs to be accessed under executor lock?

Making this conditional on `to_be_rescheduled_tasks` is also limiting the use case here I think. `ALL_TASKS_BLOCKED` is just to signal higher up that no immediate work can be done by the current executor. There can be multiple reasons for this:
- there is a blocked pipeline
- another executor has picked up the last task (i.e. there is no more tasks available for the current one) and is still executing that last task.

The latter case is also very compelling to handle with the same mechanism here, to avoid busy-spinning the current executor as it can't contribute in a meaningful to the query's completion.

Reply:
good point here, renamed it to NO_TASKS_AVAILABLE and removed the to_be_rescheduled check